plugins/modules/mso_schema_template_vrf_rp.py

```diff
@@ -0,0 +1,228 @@
+  ip:
+    description:
+    - The IP address of the RP.
+    type: str
+    aliases: [ ip_address ]
+  type:
+    description:
+    - The type of the RP.
+    type: str
+    choices: [ fabric, static ]
+  multicast_route_map_policy:
+    description:
+    - The name of the Multicast Route Map Policy.
+    - The Multicast Route Map Policy must reside in the same tenant as the tenant associated to the schema.
+    - Use O(multicast_route_map_policy="") to remove the Multicast Route Map Policy.
+    type: str
+    aliases: [ route_map_policy ]
+  state:
+    description:
+    - Use C(present) or C(absent) for adding or removing.
+    - Use C(query) for listing an object or multiple objects.
+    type: str
+    choices: [ absent, present, query ]
+    default: present
+extends_documentation_fragment: cisco.mso.modules
+"""
+
+EXAMPLES = r"""
+- name: Add a new VRF RP
+  cisco.mso.mso_schema_template_vrf_rp:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    schema: Schema 1
+    template: Template 1
+    vrf: VRF 1
+    ip: 1.1.1.1
+    type: static
+    multicast_route_map_policy: RouteMapPolicy1
+    state: present
+
+- name: Query a specific VRF RP
+  cisco.mso.mso_schema_template_vrf_rp:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    schema: Schema 1
+    template: Template 1
+    vrf: VRF 1
+    ip: 1.1.1.1
+    state: query
+  register: query_result
+
+- name: Query all VRF RPs
+  cisco.mso.mso_schema_template_vrf_rp:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    schema: Schema 1
+    template: Template 1
+    vrf: VRF 1
+    state: query
+  register: query_result
+
+- name: Remove an VRF RP
+  cisco.mso.mso_schema_template_vrf_rp:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    schema: Schema 1
+    template: Template 1
+    vrf: VRF 1
+    ip: 1.1.1.1
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import MSOModule, mso_argument_spec
+from ansible_collections.cisco.mso.plugins.module_utils.schema import MSOSchema
+from ansible_collections.cisco.mso.plugins.module_utils.template import MSOTemplate
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        schema=dict(type="str", required=True),
+        template=dict(type="str", required=True),
+        vrf=dict(type="str", required=True),
+        ip=dict(type="str", aliases=["ip_address"]),
+        type=dict(type="str", choices=["fabric", "static"]),
+        multicast_route_map_policy=dict(type="str", aliases=["route_map_policy"]),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["ip", "type"]],
```

Comment:
```suggestion
            ["state", "absent", ["ip"]],
```

`type` - not required to delete the object.

Reply:
changed, good catch thanks!